### ISSUE TITLE
Conside nfield instead of big.Int for a speedup

### ISSUE THREAD

**Issue by bmperrea** [gcash/bchd#77]
as in here https://github.com/btcsuite/btcd/pull/621

Check some benchmarks for speedup.

**Comment by zquestz** [comment]
@bmperrea Yep this is on my list to check out. I am walking through all the PRs and pulling in all the good stuff. Thanks for the heads up!

**Comment by zquestz** [comment]
@bmperrea if I were to finish up Jimmy's PR and get everything building, would you be willing to help me solve the remaining constant time issues?
